src/shared/utilities/messages.ts

```diff
@@ -45,18 +45,41 @@ export function makeFailedWriteMessage(filename: string): string {
 /**
  * Shows a non-modal error message with a button to open the log output channel.
  *
- * @returns	A promise that resolves when the button is clicked or the error is dismissed.
+ * @param message  Message text
+ * @param window  Window
+ * @param kind  Kind of message to show
+ * @param extraItems  Extra buttons shown _before_ the "View Logs" button
+ * @returns	Promise that resolves when a button is clicked or the message is
+ * dismissed, and returns the selected button text.
  */
-export async function showErrorWithLogs(message: string, window: Window = ext.window): Promise<void> {
+export async function showErrorWithLogs(
```

Comment:
Nit: All the optional args should probably be put into a new 'options' arg

Reply:
Will consider this in a followup PR, where I will also rename `showErrorWithLogs()` .